src/gui/properties/speedwidget.cpp

```diff
@@ -0,0 +1,238 @@
+#include "speedwidget.h"
+
+#include <QDateTime>
+#include <libtorrent/session_status.hpp>
+
+#include "propertieswidget.h"
+#include "core/bittorrent/session.h"
+#include "core/bittorrent/sessionstatus.h"
+#include "core/preferences.h"
+
+SpeedWidget::SpeedWidget(PropertiesWidget *parent)
+    : QWidget(parent)
+    , m_properties(parent)
+{
+    m_layout = new QVBoxLayout(this);
+    m_layout->setContentsMargins(0, 0, 0, 0);
+
+    m_hlayout = new QHBoxLayout();
+    m_hlayout->setContentsMargins(0, 0, 0, 0);
+
+    m_period_label = new QLabel("<b>" + tr("Period:") + "</b>");
+
+    m_period_combobox = new QComboBox();
+    m_period_combobox->addItem(tr("1 Minute"));
+    m_period_combobox->addItem(tr("5 Minutes"));
+    m_period_combobox->addItem(tr("30 Minutes"));
+    m_period_combobox->addItem(tr("6 Hours"));
+
+    connect(m_period_combobox, SIGNAL(currentIndexChanged(int)), this, SLOT(on_period_change(int)));
+
+    m_graphs_button = new QToolButton();
+    m_graphs_button->setText(tr("Select Graphs"));
+    m_graphs_button->setPopupMode(QToolButton::InstantPopup);
+
+    m_graphs_menu = new QMenu();
+    m_graphs_menu->addAction(tr("Total Upload"));
+    m_graphs_menu->addAction(tr("Total Download"));
+    m_graphs_menu->addAction(tr("Payload Upload"));
+    m_graphs_menu->addAction(tr("Payload Download"));
+    m_graphs_menu->addAction(tr("Overhead Upload"));
+    m_graphs_menu->addAction(tr("Overhead Download"));
+    m_graphs_menu->addAction(tr("DHT Upload"));
+    m_graphs_menu->addAction(tr("DHT Download"));
+    m_graphs_menu->addAction(tr("Tracker Upload"));
+    m_graphs_menu->addAction(tr("Tracker Download"));
+
+    graphs_menu_actions = m_graphs_menu->actions();
+    m_graphs_signal_mapper = new QSignalMapper();
+
+    for (int id = UP; id <= TRACKER_DOWN; ++id) {
+        QAction *action = graphs_menu_actions.at(id);
+        action->setCheckable(true);
+        action->setChecked(true);
+        connect(action, SIGNAL(changed()), m_graphs_signal_mapper, SLOT(map()));
+        m_graphs_signal_mapper->setMapping(action, id);
+    }
+    connect(m_graphs_signal_mapper, SIGNAL(mapped(int)), this, SLOT(on_graph_change(int)));
+
+    m_graphs_button->setMenu(m_graphs_menu);
+
+    m_hlayout->addWidget(m_period_label);
+    m_hlayout->addWidget(m_period_combobox);
+    m_hlayout->addStretch();
+    m_hlayout->addWidget(m_graphs_button);
+
+    m_plot = new QSpeedPlotView(this);
+
+    m_layout->addLayout(m_hlayout);
+    m_layout->addWidget(m_plot);
+
+    QPen greenPen;
+    greenPen.setWidthF(1.5);
+    greenPen.setColor(QColor(134, 196, 63));
+    QPen bluePen;
+    bluePen.setWidthF(1.5);
+    bluePen.setColor(QColor(50, 153, 255));
+
+    m_plot->setMaxCapacity(HOUR6_SEC);
+
+    m_plot->setGraphPen(UP, bluePen);
```

Comment:
Here you set a lot of properties on the `m_plot` object which is of type `QSpeedPlotView`.
And most of these setGraphXXX() are only called here and only once.
So it makes a lot of sense to setup these properties when constructing the `m_plot` object instead. This means you should initialize the various stuff inside `QSpeedPlotView::QSpeedPlotView() directly`
What do you say?